core/src/main/java/org/axonframework/eventhandling/SimpleEventHandlerInvoker.java

```diff
@@ -133,11 +135,40 @@ public SimpleEventHandlerInvoker(List<?> eventListeners,
                                      ParameterResolverFactory parameterResolverFactory,
                                      ListenerInvocationErrorHandler listenerInvocationErrorHandler,
                                      SequencingPolicy<? super EventMessage<?>> sequencingPolicy) {
-        this.eventListeners = new ArrayList<>(eventListeners.stream()
-                                                            .map(listener -> listener instanceof EventListener ?
-                                                                    (EventListener) listener :
-                                                                    new AnnotationEventListenerAdapter(listener, parameterResolverFactory))
-                                                            .collect(toList()));
+        this(eventListeners,
+             parameterResolverFactory,
+             new ClasspathHandlerDefinition(Thread.currentThread().getContextClassLoader()),
```

Comment:
I wouldn't change the behavior of this constructor. Instead, leave it to the `AnnotationEventListenerAdapter` to choose a default HandlerDefinition. That would allow it to create an instance based on the listener's class loader.

Reply:
Are you suggesting to leave only:
```
public SimpleEventHandlerInvoker(List<?> eventListeners,
                                     ParameterResolverFactory parameterResolverFactory,
                                     HandlerDefinition handlerDefinition,
                                     ListenerInvocationErrorHandler listenerInvocationErrorHandler,
                                     SequencingPolicy<? super EventMessage<?>> sequencingPolicy)
```
constructor?